decidim-core/app/commands/decidim/validate_upload.rb

```diff
@@ -7,9 +7,19 @@ def initialize(form)
     end
 
     def call
-      return broadcast(:invalid, @form.errors) if @form.invalid?
+      if @form.invalid?
+        remove_invalid_file
+        return broadcast(:invalid, @form.errors)
+      end
 
       broadcast(:ok)
     end
+
+    private
+
+    def remove_invalid_file
+      blob = ActiveStorage::Blob.find_signed(@form.blob)
```

Comment:
I think we should change the type of the `:blob` attribute on the form itself to `Decidim::Attributes::Blob`.

It does exactly this.

This is probably an oversight in the previous implementation (or the attribute didn't exist back then).

Reply:
@alecslupu Is there some problem implementing this change? I was just wondering because the other changes seem to be already implemented, just this one is still open.

Reply:
@ahukkanen, there is no issue. Just this PR has not been under my attention lately. also the implementation needs to be changed so that the upload flow to handle the validation as well.

Reply:
OK thanks and noted. Let me know if this still needs my attention.

Reply:
Done in 74724a2027